index.js

```diff
@@ -28,13 +29,91 @@ module.exports = function (chai, utils) {
   Assertion.overwriteMethod('eql', assertEql);
   Assertion.overwriteMethod('eqls', assertEql);
 
-  function explicitRoughly(tolerance) {
-    utils.flag(this, 'tolerance', tolerance);
+  function checkComparator(comparatorOrTolerance) {
+    if ('number' === type(comparatorOrTolerance)) {
+      return createRelativeComparator(comparatorOrTolerance);
+    } else if ('function' === type(comparatorOrTolerance)) {
+      return comparatorOrTolerance;
+    } else {
+      throw new Error('Roughly must be given a relative tolerance number or comparator function');
+    }
+  }
+
+  function explicitRoughly(comparatorOrTolerance) {
+    utils.flag(this, 'comparator', checkComparator(comparatorOrTolerance));
   }
 
   function defaultRoughly() {
-    utils.flag(this, 'tolerance', DEFAULT_TOLERANCE);
+    utils.flag(this, 'comparator', createRelativeComparator(DEFAULT_TOLERANCE));
   }
 
   Assertion.addChainableMethod('roughly', explicitRoughly, defaultRoughly);
 };
+
+// copied from deep-eql
+function simpleEqual(a, b) {
+  // Equal references (except for Numbers) can be returned early
+  if (a === b) {
+    // Handle +-0 cases
+    return a !== 0 || 1 / a === 1 / b;
+  }
+
+  // handle NaN cases
+  return (
+    a !== a && // eslint-disable-line no-self-compare
+    b !== b // eslint-disable-line no-self-compare
+  );
+}
```

Comment:
no, I meant is this the default comparator that is being used by the normal `expect().to.deep.equal()` assertion. we want to make sure that we just alter how numbers are compared, everything else should be compared the same as before.